### PR TITLE
Add Punctures analytic data and executable

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This allows to solve for any set of superposed punctures.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  };
+  struct Momentum {
+    using type = std::array<double, 3>;
+    static constexpr Options::String help{
+        "The linear momentum P of the puncture"};
+  };
+  struct Spin {
+    using type = std::array<double, 3>;
+    static constexpr Options::String help{
+        "The angular momentum S of the puncture"};
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp]: What are the bounds? Is this the dimensionless spin?

File: src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  };
+  struct Mass {
+    using type = double;
+    static constexpr Options::String help{"The puncture mass M"};
+    static double lower_bound() { return 0.; }
+  };
+  struct Momentum {
+    using type = std::array<double, 3>;
+    static constexpr Options::String help{
+        "The linear momentum P of the puncture"};
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp]: In what units?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
You can squash immediately

File: src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp (hunk 1 above)
Comment (nilsvu) [reply src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp]: These are the Bowen-York linear and angular momentum parameters in the puncture equation. They are not dimensionless. I extended the help string a bit. These are the standard parameters in the punctures ID literature (I reference the original Brandt/Brügmann paper in the dox, https://arxiv.org/abs/gr-qc/9703066).

File: src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp (hunk 2 above)
Comment (nilsvu) [reply src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp]: See comment above

File: src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp (hunk 1 above)
Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp]: Isn't there a BH associated with the puncture that has a mass, spin and angular momentum? This thing has to have some form of units... What units are you using?

Comment (nilsvu) [reply src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp]: I don't know what other units to put here than P scaling linearly in mass and S scaling quadratically. We're in c=G=1 so there's only a mass scale m, and you can choose it. P/m is dimensionless and S/m^2 is dimensionless. The punctures literature usually chooses P and S including mass factors. You would set P^x=mv for example. So I can't think of anything to add here.

Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp]: Okay, that clarifies thing. I think the important thing is to clearly define that `P^x=mv` ,etc. Can you use the dimensionless spin instead? That's what we use everywhere else and is just easier to work with. I don't know about the linear momentum, but specifying the velocity in the range `[0, 1]` naively seems more intuitive to me, but I don't have any experience with the linear momentum. Is this the coordinate momentum then? So it is bounded above by the coordinate speed of light? These are all very important details that are currently not documented and, from experience, will make it challenging for someone to use. Getting units right usually ends up being a non-negligible part of the battle.

Comment (nilsvu) [reply src/PointwiseFunctions/AnalyticData/Punctures/MultiplePunctures.hpp]: Ok, using the dimensionless spin and momentum seem more useful to me too. The bare mass of each puncture is an unambiguous parameter, so it's easy to rescale spin and momentum when comparing to other codes. Pushed a fixup.

**Review by nilsdeppe — COMMENTED** [review]
LGTM, thanks for making the change! Please go ahead and squash&rebase

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**